Release r1.1 (Fall'25 M3) - Release Candidate

alpaycetin74 opened the pull request:
#### What type of PR is this?

Add one of the following kinds:
* subproject management

#### What this PR does / why we need it:
Publication of Fall'25 M3 release candidate v0.1.0-rc.1 of VerifiedCaller APIs

#### Which issue(s) this PR fixes:
Fixes #44 

#### Special notes for reviewers:
None


#### Changelog input

```
 release-note
 - Publication of Fall'25 M3 release candidate v0.1.0-rc.1

```

#### Additional documentation 
None

hdamker commented:
@alpaycetin74 I'm a little bit confused about the planned scope of this pre-release. The scope issue #44 has no information at all, there is a release tracker for only one API ([brand-registration](https://lf-camaraproject.atlassian.net/wiki/spaces/CAM/pages/123928651/brand-registration+v0.1.0)), but there is at least already a test-definition for a second API. 

Could you please clarify within the PR description above:
* What will be the content/scope of the pre-release?
* Which of the open PRs must be merged before the release PR is ready for review? (consider to close some of the stale PRs and set PRs which are not relevant for the pre-release to "draft")

Would be great if you set the release PR to draft mode until it is actually ready for review by the Release Management team. Thanks!

alpaycetin74 commented:
Hello @hdamker , we have decided on the 2 APIs for the RC. We have updated the files accordingly. We also updated the scope issue #44, and set stale PRs to draft.
Do you think this PR is ready for review now? Thank you.

hdamker commented:
> Do you think this PR is ready for review now? Thank you.

Great progress visible!

I have done an [automated pre-check](https://github.com/hdamker/ReleaseManagement/issues/31) with our workflow under test, please see the [full report](https://github.com/hdamker/ReleaseManagement/actions/runs/16164537583/artifacts/3493534902) for details.

There are two groups of issues which are easy to fix:

* The APIs seem to be yet aligned with Commonalities 0.5 (Spring25), at least as indicated by the x-camara-commonalities field. Here is 0.6 expected, the mandatory text about `# Additional CAMARA error responses` has to be added and the XCorrelator schema (to be used also in the test definition files).
* the test definition files need to have the api-name as prefix (with the dashes)

And on a further note: the test definition files should have been provided in a separate PR, as the Release Management team can't review and approve them in details. But let's leave it like it is to safe some time.

Review of the release artifacts (CHANGELOG.md, README.md, readiness checklists, ...) will follow when a RM team member will pick up the review issue.

alpaycetin74 commented:
> > Do you think this PR is ready for review now? Thank you.
> 
> Great progress visible!
> 
> I have done an [automated pre-check](https://github.com/hdamker/ReleaseManagement/issues/31) with our workflow under test, please see the [full report](https://github.com/hdamker/ReleaseManagement/actions/runs/16164537583/artifacts/3493534902) for details.
> 
> There are two groups of issues which are easy to fix:
> 
> * The APIs seem to be yet aligned with Commonalities 0.5 (Spring25), at least as indicated by the x-camara-commonalities field. Here is 0.6 expected, the mandatory text about `# Additional CAMARA error responses` has to be added and the XCorrelator schema (to be used also in the test definition files).
> * the test definition files need to have the api-name as prefix (with the dashes)
> 
> And on a further note: the test definition files should have been provided in a separate PR, as the Release Management team can't review and approve them in details. But let's leave it like it is to safe some time.
> 
> Review of the release artifacts (CHANGELOG.md, README.md, readiness checklists, ...) will follow when a RM team member will pick up the review issue.

Thank you, I made some changes. Hopefully I got everything right.

rartych commented:
@alpaycetin74 I think in release PR the README.md needs to be updated in **Release Information** section.
As for now there is no required template - it can be short as in [DeviceIdentifier](https://github.com/camaraproject/DeviceIdentifier) or more detailed as in [QualityOnDemand](https://github.com/camaraproject/QualityOnDemand#release-information)

rartych commented on code/Test_definitions/brand-registration-deleteRegistration.feature:
```diff
@@ -0,0 +1,95 @@
+    Given an environment at "apiRoot"
+    And the resource "/brand-registration/v0.1rc1/registrations"
+    And the header "Authorization" is set to a valid access token
+    And the header "x-correlator" is set to a UUID value
```
```suggestion
    And the header "x-correlator" complies with the schema at "#/components/schemas/XCorrelator"
```

rartych commented on code/Test_definitions/brand-registration-readRegistration.feature:
```diff
@@ -0,0 +1,103 @@
+    Given an environment at "apiRoot"
+    And the resource "/brand-registration/v0.1rc1/registrations"
+    And the header "Authorization" is set to a valid access token
+    And the header "x-correlator" is set to a UUID value
```
```suggestion
    And the header "x-correlator" complies with the schema at "#/components/schemas/XCorrelator"
```

rartych commented on code/Test_definitions/brand-registration-updateRegistration.feature:
```diff
@@ -0,0 +1,116 @@
+    And the resource "/brand-registration/v0.1rc1/registrations"
+    And the header "Content-Type" is set to "application/json"
+    And the header "Authorization" is set to a valid access token
+    And the header "x-correlator" is set to a UUID value
```
```suggestion
    And the header "x-correlator" complies with the schema at "#/components/schemas/XCorrelator"
```

rartych commented on code/API_definitions/brand-registration.yaml:
```diff
@@ -1,741 +1,780 @@
+                  status:
+                    enum:
+                      - 401
+                  code:
+                    enum:
+                      - AUTHENTICATION_REQUIRED
+                      - UNAUTHENTICATED
+          examples:
+            GENERIC_401_AUTHENTICATION_REQUIRED:
+              description: New authentication is needed, authentication is no longer valid
+              value:
+                status: 401
+                code: AUTHENTICATION_REQUIRED
+                message: New authentication is required.
+            GENERIC_401_UNAUTHENTICATED:
+              summary: Unauthenticated
+              description: Request cannot be authenticated
```
401 AUTHENTICATION_REQUIRED code  should be removed - see https://github.com/camaraproject/Commonalities/pull/429

rartych commented:
Final review remarks added.
After reviewing them please rebase this PR here with the main branch (sync the CODEOWNERS update).

rartych reviewed (dismissed):
Looks good from the Release Management perspective.

GillesInnov35 reviewed (approved):
LGTM

stroncoso-quobis reviewed (approved):
LGTM!

alpaycetin74 commented:
Could you re-review as I made a new commit to change the optionality of request parameters, as we agreed yesterday ? Also updated a test related to it. Thank you @rartych @stroncoso-quobis @GillesInnov35

GillesInnov35 reviewed (approved):
Thanks a lot Cetin for those updates.

rartych commented:
@alpaycetin74 If I can help you with producing the release "CAMARA way" we can have a short call this afternoon.

hdamker commented:
> I think the "Latest" tag is not necessary. Once the final release is published (M4) Github will present it as the latest.

This tag is indeed not needed, it's even confusing. I just deleted it.

`https://github.com/camaraproject/VerifiedCaller/releases/latest` is automatically resolved by GitHub:
* If there are public (= non pre-release) releases, it will automatically show either the latest done or the one marked as "latest" in the release options (see for example  https://github.com/camaraproject/QualityOnDemand/releases/latest)
* If there is not yet a public release (only pre-release) it will show the list of releases

A tag "Latest" as created would be something completly different.
